signing/src/main/java/tech/pegasys/web3signer/signing/config/SecpArtifactSignerProviderAdpater.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.signing.config;
+
+import static org.web3j.crypto.Keys.getAddress;
+import static tech.pegasys.web3signer.signing.util.IdentifierUtils.normaliseIdentifier;
+
+import tech.pegasys.web3signer.signing.ArtifactSigner;
+import tech.pegasys.web3signer.signing.ArtifactSignerProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class SecpArtifactSignerProviderAdpater implements ArtifactSignerProvider {
+  private static final Logger LOG = LogManager.getLogger();
+  private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+  private final Map<String, ArtifactSigner> signers = new HashMap<>();
+  private final ArtifactSignerProvider signerProvider;
+
+  public SecpArtifactSignerProviderAdpater(final ArtifactSignerProvider signerProvider) {
+    this.signerProvider = signerProvider;
+  }
+
+  @Override
+  public Future<Void> load() {
+    return executorService.submit(
+        () -> {
+          LOG.debug("Adding eth1 address for eth1 keys");
+
+          signerProvider
+              .availableIdentifiers()
+              .forEach(
+                  (publicKey) -> {
+                    signerProvider
+                        .getSigner(publicKey)
+                        .ifPresent(
+                            signer ->
+                                signers.putIfAbsent(
+                                    normaliseIdentifier(getAddress(publicKey)), signer));
+                  });
+
+          return null;
+        });
+  }
+
+  @Override
+  public Optional<ArtifactSigner> getSigner(String identifier) {
```

Comment:
nit: coming back to this code, it's not clear what this `String identifier` represents - I had to follow from EthSignTransactionResultProvider down to here and I guess it's an ethereum address based on the load method above. Maybe a better name at the RPC level would help.
Separately, EthSecpArtifactSigner uses a public key instead which makes this more confusing. 
I think it's probably a wider issue of using a String type in the interface so maybe out of scope for this PR.